src/Servers/HttpSys/test/FunctionalTests/Listener/ResponseHeaderTests.cs

```diff
@@ -221,6 +220,28 @@ public async Task ResponseHeaders_HTTP10KeepAliveRequest_Gets11Close()
             HttpResponseMessage response = await responseTask;
             response.EnsureSuccessStatusCode();
             Assert.Equal(new Version(1, 1), response.Version);
+            Assert.Null(response.Headers.ConnectionClose);
```

Comment:
Ideally, we'd verify that not only was the `Connection: close` header not sent, but that the underlying socket wasn't closed. The easiest way to verify this would probably be to hook [SocketsHttpHandler.ConnectCallback](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.socketshttphandler.connectcallback?view=net-8.0) and verify that it was only called once for 2 requests.

Reply:
Good point, added the verification